src/ImageSharp/MetaData/Profiles/ICC/TagDataEntries/IccLut16TagDataEntry.cs

```diff
@@ -62,17 +62,14 @@ public IccLut16TagDataEntry(float[,] matrix, IccLut[] inputValues, IccClut clutV
             : base(IccTypeSignature.Lut16, tagSignature)
         {
             Guard.NotNull(matrix, nameof(matrix));
-            Guard.NotNull(inputValues, nameof(inputValues));
-            Guard.NotNull(clutValues, nameof(clutValues));
-            Guard.NotNull(outputValues, nameof(outputValues));
 
             bool is3By3 = matrix.GetLength(0) == 3 && matrix.GetLength(1) == 3;
             Guard.IsTrue(is3By3, nameof(matrix), "Matrix must have a size of three by three");
 
             this.Matrix = this.CreateMatrix(matrix);
-            this.InputValues = inputValues;
-            this.ClutValues = clutValues;
-            this.OutputValues = outputValues;
+            this.InputValues = inputValues ?? throw new ArgumentNullException(nameof(inputValues));
```

Comment:
I really prefer using Guard instead of the null-coalescing operator.

Reply:
They look nicer, but it is worth the expense of an extra call and convoluted stack when an exception is thrown? I would advocate that we eventually replace all Guard methods with methods that throw inline so the user knows exactly where the problem is.